### ISSUE TITLE
logical bitwise in where clause not working

### ISSUE THREAD

**Issue by amirjalali1** [dotnet/efcore#5755]
### Steps to reproduce

```
    **[Flags]**
    public enum StudentType
    {
        Default,
        A,
        B
    }
```

```
    public class Student
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public StudentType StudentType { get; set; }
    }
```
### The issue

The following query should return all students of type B. But the generated tsql script doesn't have the required where clause.

```
   using(var dbCtx = new MyDbContext())
            {
             var students =   dbCtx.Students.Where(s => s.StudentType.HasFlag(StudentType.B)).ToList();
            }
```

even if i write the linq query like this

`var students = dbCtx.Students.Where(s => (s.StudentType & StudentType.B) !=0).ToList();`

the generated tsql  still dose not have the where clause
### Further technical details

EF Core version: (rc2)
Operating system: windows 10 
Visual Studio version: (e.g. VS 2015)

**Comment by smitpatel** [comment]
For the first query - `Enum.HasFlag` translation to Sql is not yet supported.
For the second query - Translation to Sql was implemented as per #4255 which will be available in next release.
For both the queries, even though there is no where clause in Sql, it will be filtered on client side.

**Comment by rowanmiller** [comment]
Enum.HasFlag is tracked by https://github.com/aspnet/EntityFramework/issues/2852 (and there is a PR open to support it https://github.com/aspnet/EntityFramework/pull/5576)
